fe/fe-core/src/main/java/org/apache/doris/datasource/PooledHiveMetaStoreClient.java

```diff
@@ -62,49 +64,56 @@ public List<String> getAllDatabases() {
         try (CachedClient client = getClient()) {
             return client.client.getAllDatabases();
         } catch (Exception e) {
-            throw new RuntimeException(e);
+            throw new HMSClientException("failed to get all database from hms client", e);
         }
     }
 
     public List<String> getAllTables(String dbName) {
         try (CachedClient client = getClient()) {
             return client.client.getAllTables(dbName);
         } catch (Exception e) {
-            throw new RuntimeException(e);
+            throw new HMSClientException("failed to get all tables for db %s", e, dbName);
         }
     }
 
     public boolean tableExists(String dbName, String tblName) {
         try (CachedClient client = getClient()) {
             return client.client.tableExists(dbName, tblName);
         } catch (Exception e) {
-            throw new RuntimeException(e);
+            throw new HMSClientException("failed to check if table %s in db %s exists", e, tblName, dbName);
+        }
+    }
+
+    public List<String> listPartitionNames(String dbName, String tblName) {
+        try (CachedClient client = getClient()) {
+            return client.client.listPartitionNames(dbName, tblName, MAX_LIST_PARTITION_NUM);
```

Comment:
Could it cause wrong query result if the partitions in Hive are more than MAX_LIST_PARTITION_NUM?

Reply:
I tested and yes, it will only return partitions up to the limit, and the result will be wrong.
I will find a way to handle this later.